### PR TITLE
[Azure Search] Remove inheritdoc and improve SearchCredentials documentation

### PR DESCRIPTION
**Pull request by bitsforthought**
`<inheritdoc />` is not supported by the .NET API browser. As a workaround, I'm copying documentation instead. Also adding more detail to `SearchCredentials` per feedback from Heidi.

Once this is in the preview SDK, I'll merge it back to the GA SDK.

FYI @HeidiSteen @Yahnoosh @mhko @jeji1101

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by HeidiSteen** [comment]
minor inconsistancy -- not an error, just noting it:  "otherwise, false" vs "false otherwise"
Customizations/Search/Models/ExtensibleEnum.cs 
Lines 73 and 88

On the short parameter descriptions, some have periods and some don't
Customizations/DataSources/DataSourceOperations.Customization.cs
Customizations/Indexers/IndexersOperations.Customization.cs 
Customizations/Indexes/IndexesOperations.Customization.cs 


unfamiliar HTML code -- I've never seen <c></c>
There are 9 instances -- maybe do a search on the PR? 
Here is one:
Customizations/Indexers/IndexersOperations.Customization.cs
Line 64
<c>true</c> if the indexer exists; <c>false</c> otherwise.

**Comment by HeidiSteen** [comment]
I approve the PR as long as the <c> is valid HTML syntax.

**Comment by brjohnstmsft** [comment]
@HeidiSteen Ironically, it's the tags you didn't recognize that turned out to be the correct ones. This is technically .NET XML docs, not HTML, which can fool me sometimes. Not all HTML tags are supported by the .NET API Browser. In this case, `<c></c>` *is* supported, because it's how references or emphasis is indicated in .NET XML docs. Looking at how the existing docs are rendered on docs.microsoft.com, it appears that `<b></b>` is not supported, so I'm going to replace all the b's with c's.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/DataSources/DataSourceOperations.Customization.cs
Comment (UD1412) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/DataSources/DataSourceOperations.Customization.cs]: missing 'returns' tag

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/Indexers/IndexersOperations.Customization.cs
Comment (UD1412) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/Indexers/IndexersOperations.Customization.cs]: missing a 'returns' tag

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/Indexes/IndexesOperations.Customization.cs
Comment (UD1412) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/Indexes/IndexesOperations.Customization.cs]: same as above

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/SynonymMaps/SynonymMapsOperations.Customization.cs
Comment (UD1412) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/SynonymMaps/SynonymMapsOperations.Customization.cs]: same as above.

**Review by UD1412 — CHANGES_REQUESTED** [review]

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/Indexers/IIndexersOperations.Customization.cs
Comment (UD1412) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/Indexers/IIndexersOperations.Customization.cs]: The tag is set to 'returns' at quite a few places. Please correct these.

**Review by UD1412 — APPROVED** [review]

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/Indexers/IIndexersOperations.Customization.cs
Comment (brjohnstmsft) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/Customizations/Indexers/IIndexersOperations.Customization.cs]: It looks like an issue with the generated code that copied over to the custom code. It's supposed to be `<returns>` everywhere. I'll fix it in the custom code. In the meantime, I've filed an issue on AutoRest: https://github.com/Azure/autorest/issues/2957

**Comment by brjohnstmsft** [comment]
@UD1412 Would you mind approving using your other GitHub account, @updixit? The PR needs to be approved by an account with merge permissions to the `search-preview` branch.

**Comment by shahabhijeet** [comment]
@brjohnstmsft @HeidiSteen is this ready to be merged.
We are in middle of upgrading our repo, so if this PR is ready to be merged, this has to be approved and merged today, else we will close this PR and you will need to open a new PR against the upgraded repo.

**Comment by HeidiSteen** [comment]
#sign-off

**Comment by updixit** [comment]
#sign-off

**Review by mhko — APPROVED** [review]

**Comment by UD1412** [comment]
#sign-off

**Comment by shahabhijeet** [comment]
@azuresdkci retest this please

**Pull request merged**
